panel/pane/plotly.py

```diff
@@ -33,6 +43,24 @@ def applies(cls, obj):
                 hasattr(obj, 'to_plotly_json') or (isinstance(obj, dict)
                                                    and 'data' in obj and 'layout' in obj))
 
+    def __init__(self, object=None, **params):
+        import plotly.graph_objs as go
+        super(Plotly, self).__init__(object, **params)
+
+        if type(object) is go.Figure:
+
+            # Monkey patch the message stubs used by FigureWidget.
+            # We only patch `Figure` objects (not subclasses like FigureWidget) so
+            # we don't interfere with subclasses that override these methods.
+            fig = object
+            fig._send_addTraces_msg = partial(self._update_figure, fig)
+            fig._send_moveTraces_msg = partial(self._update_figure, fig)
+            fig._send_deleteTraces_msg = partial(self._update_figure, fig)
+            fig._send_restyle_msg = partial(self._update_figure, fig)
+            fig._send_relayout_msg = partial(self._update_figure, fig)
+            fig._send_update_msg = partial(self._update_figure, fig)
+            fig._send_animate_msg = partial(self._update_figure, fig)
```

Comment:
These can all use `lambda: self.param.trigger('object')` or similar.

Reply:
I think this also needs to happen not only on the constructor but whenever a user updates the ``object``. I think we should probably add a callback on changes to ``.object`` which calls `_to_figure` and caches the created figure. Something like:

```python
@param.depends('object')
def _update_figure(self):
     if self.object is not None and id(self.object) not in self._figure:
         fig = self._to_figure(self.object)
         self._figure = {id(self.object): fig}
         fig._send_addTraces_msg = lambda: self.param.trigger('object')
         ...
```

Reply:
That all makes sense. Will do